ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -440,7 +452,7 @@ func resourceIBMPIInstanceRead(ctx context.Context, d *schema.ResourceData, meta
 	}
 	d.Set("min_processors", powervmdata.Minproc)
 	d.Set(helpers.PIInstanceProgress, powervmdata.Progress)
-	if powervmdata.StorageType != nil {
+	if powervmdata.StorageType != nil && *powervmdata.StorageType != "" {
```

Comment:
If the value is empty you need to set the attr as "". So if condition is not required here.

Reply:
Set them "" if not available